include/bm/bm_sim/data.h

```diff
@@ -387,12 +387,28 @@ class Data {
     return lhs.value == rhs.value;
   }
 
+  //! NC
+  template<typename T,
+           typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
+  friend bool operator==(const Data &lhs, const T &rhs) {
+    assert(lhs.arith);
+    return lhs.value.convert_to<T>() == rhs;
+  }
+
   //! NC
   friend bool operator!=(const Data &lhs, const Data &rhs) {
     assert(lhs.arith && rhs.arith);
     return !(lhs == rhs);
   }
 
+  //! NC
+  template<typename T,
+           typename std::enable_if<std::is_integral<T>::value, int>::type = 0>
+  friend bool operator!=(const Data &lhs, const T &rhs) {
+    assert(lhs.arith);
+    return !(lhs == rhs);
+  }
+
```

Comment:
Unless you think this is necessary, I'd rather let the caller do the conversion themselves
```c++
x == data.to<uint64_t>
```